bucket.go

```diff
@@ -54,26 +57,26 @@ func (b *bucket) peers() []PeerInfo {
 	return ps
 }
 
-// returns the "minimum" peer in the bucket based on the `lessThan` comparator passed to it.
-// It is NOT safe for the comparator to mutate the given `PeerInfo`
-// as we pass in a pointer to it.
+// findFirst returns the first peer in the bucket that satisfies the given predicate.
+// It is NOT safe for the predicate to mutate the given `PeerInfo` as we pass in a pointer to it.
 // It is NOT safe to modify the returned value.
-func (b *bucket) min(lessThan func(p1 *PeerInfo, p2 *PeerInfo) bool) *PeerInfo {
-	if b.list.Len() == 0 {
-		return nil
+func (b *bucket) findFirst(p func(p *PeerInfo) bool) *PeerInfo {
```

Comment:
I'd still use a comparison function like `min` that determines which peer is more "evictable" than another peer. Then we go through the bucket and find the most evictable peer and check if it is actually evictable.

Previously: Find oldest time, then check if time is too old
This PR: Find first replaceable
Suggestion: comparison function `if p1.replaceable return p1; if p2.replaceable return p2; return p1`. Evaluation function `return p.replaceable`. It's really easy to plug in the older code here if required.

Note: if we wanted to enable stable sorting we would have to compare peerIDs if both results were equal (e.g. added at the same time or both replaceable), but since we just need one of them it shouldn't really matter.

Reply:
I've made this change. Let me know what you think. And I agree, we don't need a stable sort for now.